app/models/address.rb

```diff
@@ -98,7 +98,7 @@ def search(_location, components, postcode)
       address = Address.find_by('lower(street_address) IN (?)', components.map(&:downcase))
       return address if address
 
-      ukpc = UKPostcode.parse(postcode)
+      ukpc = UKPostcode.parse(postcode.to_s)
 
       if ukpc.full_valid?
```

Comment:
For example, if `UKPostcode.parse(nil) == nil`, this could become something like:

```rb
ukpc = UKPostcode.parse(postcode)

if ukpc.present? && ukpc.full_valid?
```

If it throws an error instead (which I'm guessing is what's happening here given your approach), we could do:

```rb
ukpc = UKPostcode.parse(postcode) if postcode.present?

if ukpc.present? && ukpc.full_valid?
```

Reply:
its a fairly common pattern in ruby to `thing.to_s` if you are unsure that thing may not be a string. I find the whole `thing&.full_valid` bit to be a bit hacky as its basically expanded into `(thing && thing.full_valid?)`

Reply:
Not sure I follow why that expansion makes it hacky?